packages/hash/frontend/src/pages/[shortname]/types/entity-type/[entity-type-id].page/definition-tab/property-list-card/shared/property-type-form/expected-value-selector/custom-expected-value-builder/array-expected-value-builder/array-min-max-items.tsx

```diff
@@ -0,0 +1,302 @@
+
+  const setInputSize = useCallback(() => {
+    setMinItemsWidth(minItemsInputRef.current?.value.length ?? 1);
+    setMaxItemsWidth(maxItemsInputRef.current?.value.length ?? 1);
+  }, []);
+
+  useEffect(() => {
+    setInputSize();
+  }, [setInputSize]);
+
+  return (
+    <Box display="flex" gap={1.25}>
+      <Box display="flex">
+        <Box
+          sx={{
+            display: "inline-flex",
+            alignItems: "center",
+            py: 0.25,
+            px: 1,
+            background: ({ palette }) => palette.gray[80],
+            borderBottomLeftRadius: 30,
+            borderTopLeftRadius: 30,
+          }}
+        >
+          <Typography
+            variant="smallCaps"
+            sx={{
+              fontSize: 11,
+              color: ({ palette }) => palette.gray[30],
+            }}
+          >
+            Min
+          </Typography>
+        </Box>
+        <Box
+          sx={{
+            display: "inline-flex",
+            alignItems: "center",
+            py: 0.25,
+            pr: 1,
+            pl: 0.75,
+            background: ({ palette }) => palette.gray[90],
+            borderBottomRightRadius: 30,
+            borderTopRightRadius: 30,
+          }}
+        >
+          <ItemInput
+            {...minItemsController.field}
+            inputRef={minItemsInputRef}
+            width={`${minItemsWidth || 1}ch`}
+            onChange={(evt) => {
+              const target = evt.target as HTMLInputElement;
+
+              const min = Number.isNaN(target.valueAsNumber)
+                ? 0
+                : Math.max(0, target.valueAsNumber);
+
+              if (min > maxItemsController.field.value) {
+                setValue(
+                  `flattenedCustomExpectedValueList.${arrayId}.data.maxItems`,
+                  min,
+                  { shouldDirty: true },
+                );
+              }
+              minItemsController.field.onChange(min);
+
+              setImmediate(() => {
+                setInputSize();
+              });
+            }}
+          />
+        </Box>
+      </Box>
+
+      <Box
+        display="flex"
+        onMouseEnter={() => setHovered(true)}
+        onMouseLeave={() => setHovered(false)}
+      >
+        <Box
+          sx={{
+            display: "inline-flex",
+            alignItems: "center",
+            py: 0.25,
+            px: 1,
+            background: ({ palette }) => palette.gray[80],
+            borderBottomLeftRadius: 30,
+            borderTopLeftRadius: 30,
+          }}
+        >
+          <Typography
+            variant="smallCaps"
+            sx={{
+              fontSize: 11,
+              color: ({ palette }) => palette.gray[30],
+            }}
+          >
+            Max
+          </Typography>
+        </Box>
+        <Box
+          sx={{
+            display: "inline-flex",
+            alignItems: "center",
+            py: 0.25,
+            px: 1,
+            pl: 0.75,
```

Comment:
Nitpick: Instead of using `px - pl`, using `pr - pl` sounds more natural to me. But this is totally fine imo.

Reply:
Changed in 6a9446eb2bca9c4d0d5567d6abe29ca2862496d2